src/osmet-pack

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -euo pipefail
+
+if [ ! -f /etc/cosa-supermin ]; then
+    dn=$(dirname "$0")
+    # shellcheck source=src/cmdlib.sh
+    . "${dn}"/cmdlib.sh
+
+    img_src=$1; shift
+    osmet_dest=$1; shift
+    checksum=$1; shift
+    coreinst=${1:-${OSMET_PACK_COREOS_INSTALLER:-}}
+
+    workdir=$(pwd)
+    TMPDIR=tmp/tmp-osmet-pack
+    rm -rf "${TMPDIR}"
+    mkdir -p "${TMPDIR}"
+
+    if [[ $img_src == *.gz || $img_src == *.xz ]]; then
+        img="$(basename "$img_src")"
+        fatal "Cannot pack osmet from $img; not an uncompressed image"
+    fi
+
+    set -- "${TMPDIR}/osmet.bin" "${checksum}"
+    if [ -n "${coreinst:-}" ]; then
+        cp "${coreinst}" "${TMPDIR}/coreos-installer"
+        set -- "$@" "${TMPDIR}/coreos-installer"
+    fi
+
+    runvm -drive "if=virtio,id=coreos,format=raw,readonly=on,file=${img_src}" -- \
```

Comment:
I was thinking about this runvm stuff recently and I realized that we could actually switch to a two-phased system where we first build qemu/metal image, and then all `runvm` tasks after that simply boot that image with an Ignition config - making us much more self-hosting.

(The downside of this is that for RHCOS we can't use 9p but that problem will really, hopefully soon be solved with [virtiofs](https://virtio-fs.gitlab.io/)).

Reply:
To elaborate on this, we wouldn't have a `runvm` API internally distinct from `buildextend-qemu`.

Reply:
Would be interesting. One thing we lose though is that supermin automatically imports all the packages from the build environment. Obviously we could do something similar with Ignition/a systemd service fetching the files we want, but it's nice that it works transparently with supermin.

It would be a good fit for osmet though since we normally want to use the coreos-installer from the FCOS node in that case. So in the developer case, we only have to upload the custom coreos-installer really.